docs/guide/basic-usage.md

```diff
@@ -40,4 +40,139 @@ $query->select(['name', 'status'])
 // execute the query
 $rows = $query->all();
 ```
+## Aggregation
 
+Originally MongoDB [aggregate](https://docs.mongodb.com/manual/aggregation/) is accepting a pipeline. Pipeline is a group of
+stages which is executed sequentially. Each stage does operation on a previous stage results to filter the result.
+
+You can use Yii aggregate method to achieve functions like grouping results by one or more field 
+or even building more complex aggregation pipeline.
+  
+### Group by One or More Fields
+
+The following example display how you can group books by `$authorId` field.
+
+```php
+$collection = Yii::$app->mongodb->getCollection('books');
+
+$authors = $collection->aggregate([
+    [
+        '$group' => [
+            '_id' => '$authorId',
+        ],
+    ],
+]);        
+```
+
+### Group by Multiple Fields
```

Comment:
Examples are too verbose.
`yii2-mongodb` simply calls `aggregate` command composing JSON from PHP array.
All nuances should remain at mongodb.org documentationm otherwise such docs updates will continue untill it all been copied to this repo.